File: CodeForces/Codeforces Round 923 (Div. 3)/1927A_Make_it_white.py
# Function to find the minimum length of segment needed to paint to cover all black cells
def min_paint_segment(n, s):
    left_black = s.find('B')  # Find the first black cell from the left
    right_black = s.rfind('B')  # Find the first black cell from the right
    
    # Calculate the length of the segment between the first and last black cell
    min_segment_length = right_black - left_black + 1
    
    return min_segment_length

# Read the number of test cases
t = int(input())

# Iterate through each test case
for _ in range(t):
    # Read the length of the strip and the strip itself
    n = int(input())
    s = input().strip()
    
    # Find and output the minimum length of segment needed to paint
    print(min_paint_segment(n, s))




""" 
8
6
WBBWBW
1
B
2
WB
3
BBW
4
BWWB
6
BWBWWB
6
WWBBWB
9
WBWBWWWBW


 """

File: CodeForces/Codeforces Round 923 (Div. 3)/1927A_Make_it_white.cpp
#include <bits/stdc++.h>
#include <iostream>
#include <iomanip>
#include <cmath>
#include <string>
#include <cstdlib>
#define pi 3.14159
#define forn(i, n) for (int i = 0; i < int(n); i++)

#define nl endl
#define ll long long
#define ld long double
#define ull unsigned long long
#define mod 90000007
#define fs(n) fixed << setprecision(int(n))
#define s(n) setprecision(int(n))
using namespace std;


// Function to find the minimum length of segment needed to paint to cover all black cells
int min_paint_segment(int n, string s) {
    int left_black = s.find('B');  // Find the first black cell from the left
    int right_black = s.rfind('B');  // Find the first black cell from the right
    
    // Calculate the length of the segment between the first and last black cell
    int min_segment_length = right_black - left_black + 1;
    
    return min_segment_length;
}

int main() {
    // Read the number of test cases
    int t;
    cin >> t;
    
    // Iterate through each test case
    for (int i = 0; i < t; ++i) {
        // Read the length of the strip and the strip itself
        int n;
        cin >> n;
        string s;
        cin >> s;
        
        // Find and output the minimum length of segment needed to paint
        cout << min_paint_segment(n, s) << endl;
    }
    
    return 0;
}